py/dml/serialize.py

```diff
@@ -462,14 +463,17 @@ def error_out(exc, msg):
             stmts.append(ctree.mkInline(site, 'goto _exit;'))
             return stmts
         if isinstance(real_type, TStruct):
+            (tmp_out_decl, tmp_out_ref) = declare_variable(
+                site, "_tmp_out", TPtr(real_type),
+                ctree.mkNew(site, real_type))
+            tmp_out_decl.toc()
+            cleanup.append(ctree.mkDelete(site, tmp_out_ref))
             statements = []
             imm_attr_decl, imm_attr_ref = declare_variable(
                 site, "_imm_attr", attr_value_t)
-            tmp_out_decl, tmp_out_ref = declare_variable(
-                site, "_tmp_out", real_type)
             for (i, (name, val_type)) in enumerate(real_type.members.items()):
                 index = ctree.mkIntegerConstant(site, i, False)
-                val_ref = ctree.mkSubRef(site, tmp_out_ref, name, ".")
+                val_ref = ctree.mkSubRef(site, tmp_out_ref, name, "->")
```

Comment:
As a complete sidenote;
Doesnt mkSubRef's argument know about it's typing? Why do we have to specify which access operator to use?
And sidenote sidenote;
If we _do_ have to specify the operator, shouldn't this be two separate functions (since we only have two valid operators)?